spock-tapestry/tapestry.gradle

```diff
@@ -4,23 +4,23 @@ ext.displayName = "Spock Framework - Tapestry Module"
 
 description = "Spock's Tapestry Module provides support for testing Tapestry 5 based applications."
 
-def tapestryVersion54 = "5.4.3"
+def tapestryVersion57 = "5.7.2"
 
-def useTapestry54 = javaVersion >= 1.8
+def useTapestry57 = javaVersion >= 1.8
 
-def tapestryVersionTest = useTapestry54 ? tapestryVersion54 : "5.2.6"
+def tapestryVersionTest = useTapestry57 ? tapestryVersion57 : "5.2.6"
```

Comment:
So with these changes `spock-tapestry` would support 5.2.x and 5.7.2+ but not the versions in between?

Reply:
Actually, based on https://github.com/spockframework/spock/issues/1312#issuecomment-826810060, the support for earlier releases of Tapestry is no longer necessary. I'll commit and update.

Reply:
Why not?
If you mean
> Current and previous tapestry-spock versions cover previous versions of Tapestry.

then I don't think that makes sense.
You cannot use the "current or previous" version of `spock-tapestry` with a newer version of Spock.
If spock should support the old Tapestry versions, then it must support them, probabyl with different variants.

Reply:
What I understood @thiagohp meant is that users of Tapestry up to release 5.6.+ would be able to use Spock up to release 2.0-M5, whereas Tapestry 5.7+ users would have to use the first Spock release including this pull request, or later.
In my opinion, that's a reasonable suggestions. I imagine it would work for most users of Tapestry and Spock.

Reply:
@vjlamp: You understood correctly. :+1:

Reply:
Please let me know if I can provide additional information to help with the decision making. Continuing with `spock-tapestry`would be ideal for Tapestry users upgrading to release 5.7. Moving the code to `tapestry-spock` subproject technically possible, of course, but would be a source of confusion. I'm also not sure about the legal/license prerequisites. As a third option, I could roll my own Spock Extension. This is probably the least attractive option for the community. Please let me know how you intend to proceed.

Reply:
Unfortunately this change won't make it into Spock 2.0 as we have been in feature freeze for the upcoming release. What I've been asking in https://github.com/spockframework/spock/issues/1312#issuecomment-826974578 was whether you'd be willing to adopt that module and maintain it outside of the spock repository. The extension could be written in such a way that it works for multiple spock versions, and also could support different tapestry versions as well.

Reply:
If they don't want the extension in their repository, you could either create a dedicated organization+repository yourself, or just create a repository under your username. 
Either way we can list the new location here https://github.com/spockframework/spock/wiki/Third-Party-Extensions

Reply:
The new `tapestry-spock` sub-project - essentially a copy of `spock-tapestry` was added to Tapestry yesterday and is scheduled for release with version 5.7.3. It works as a drop-in replacement.

See https://github.com/apache/tapestry-5/tree/master/tapestry-spock for details.

I'm glad I didn't have to create my own Spock Extension. Thank you, @leonard84 and @thiagohp, for playing ball.

Reply:
Great, I also added it to https://github.com/spockframework/spock/wiki/Third-Party-Extensions

Reply:
@vjlamp maybe give me a ping here or in the issue so that I can announce it via spock twitter.